### PR TITLE
[nms,agw,orc8r] Adding MME_IP to eNB configs

### PR DESCRIPTION
**Pull request by hakeen0618**
Signed-off-by: HakeenHan hanhufeng@baicells.com
## Summary

- Adding MME_IP value to eNB  configs
- MME_IP parameter is used to set enb  in the nms UI .

## Test Plan

- make precommit
- running locally with eNB configured manually.

## Additional Information

- [ ] This change is backwards-breaking

This adds mme_ip field for managed configs. however we don't have any of these registered yet on any of the environments so we should be good.

### CODE REVIEW

**Comment by hakeen0618** [comment]
NMS: 
# #8124 
Screenshot from 2021-07-16 01-06-53

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 1]
```diff
@@ -2286,6 +2286,10 @@ definitions:
           - 'Baicells ID TDD/FDD'
           - 'NuRAN Cavium OC-LTE'
         x-nullable: false
+      mme_ip:
+        type: string
+        format: ipv4
+        example: '10.0.2.1'
```
Comment (uri200) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2292]: maybe you can set a default value to empty string
```
default: ""
```

**Comment by amarpad** [comment]
@ardzoht why can't this just be derived from the AGW?

File: lte/gateway/python/magma/enodebd/device_config/configuration_init.py [hunk 2]
```diff
@@ -199,6 +199,8 @@ def _get_enb_config(
             tac = enb_config.tac
             bandwidth_mhz = enb_config.bandwidth_mhz
             cell_id = enb_config.cell_id
+            mme_address = enb_config.mme_ip
+            mme_port = 36412
```
Comment (ardzoht) [inline lte/gateway/python/magma/enodebd/device_config/configuration_init.py]: What is this value from? Are you adding configuration value for this as well?

File: nms/app/packages/magmalte/app/components/lte/AddEditEnodebDialog.js [hunk 3]
```diff
@@ -83,6 +83,9 @@ export default function AddEditEnodebDialog(props: Props) {
   );
   const [serialId, setSerialId] = useState<string>(editingEnodeb?.serial || '');
   const [tac, setTac] = useState(String(editingEnodeb?.config.tac || ''));
+  const [mme_ip, setMmeIP] = useState(
+    String(editingEnodeb?.config.mme_ip || '10.0.2.1'),
```
Comment (ardzoht) [inline nms/app/packages/magmalte/app/components/lte/AddEditEnodebDialog.js]: Defaulting to @andreilee for this, but it seems iffy to me to define a hard coded string here, do we have a constants file for this to be defined?

**Review by ardzoht — CHANGES_REQUESTED** [review]

**Comment by ardzoht** [comment]
Requested some changes, may I ask what is the long term plan to add this config value? With this change we will add NMS support to stream this value and display it, but it doesn’t change any of the actual device config, is this planned for configuring managed radios?

if you could add some context on the Summary section that would be good to follow up from there, thanks!

**Review by amarpad — CHANGES_REQUESTED** [review]
MME IP should just be derived from the AGW the radio is getting attached to.

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 1 above)
Comment (hakeen0618) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2292]: This example is only use in swagger api response doc, and I set it reference  the ip_addr configuration as following picture in the history.
ip

File: nms/app/packages/magmalte/app/components/lte/AddEditEnodebDialog.js (hunk 3 above)
Comment (hakeen0618) [reply nms/app/packages/magmalte/app/components/lte/AddEditEnodebDialog.js]: ok , i will make changes and git push.
mme_nms_default

File: lte/gateway/python/magma/enodebd/device_config/configuration_init.py (hunk 2 above)
Comment (hakeen0618) [reply lte/gateway/python/magma/enodebd/device_config/configuration_init.py]: yes, I  use DEFAULT_S1_PORT and replace it.
mme_hardcode

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 1 above)
Comment (hakeen0618) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2292]: > Requested some changes, may I ask what is the long term plan to add this config value? With this change we will add NMS support to stream this value and display it, but it doesn’t change any of the actual device config, is this planned for configuring managed radios?
> 
> if you could add some context on the Summary section that would be good to follow up from there, thanks!

OK, I will add the Summary section.

**Comment by sourabh-nanoti** [comment]
> Requested some changes, may I ask what is the long term plan to add this config value? With this change we will add NMS support to stream this value and display it, but it doesn’t change any of the actual device config, is this planned for configuring managed radios?
> 
> if you could add some context on the Summary section that would be good to follow up from there, thanks!
@ardzoht I think the enodebd has been changed so that TR069 sends this parameter to the ENB 
@hakeen0618  Can you please add some clarification here as requested by @ardzoht ? Thanks.

**Comment by sourabh-nanoti** [comment]
> MME IP should just be derived from the AGW the radio is getting attached to.

cc  @talkhasib

**Comment by talkhasib** [comment]
I agree that MME IP should be derived from the AGW managing the eNB. Not through NMS.
But, there is value in this if we ever end up moving enodebd to cloud. Thus, MME IP is different for enodebd host.

**Comment by talkhasib** [comment]
But, in the current context. This is not necessary and adds confusion.
There are no plans to move enodebd out of the AGW. Thus, I support closing this PR.
This was a good exercise for the authors to get a parameter added e2e. Lets discuss what other parameters make better sense.

**Review by talkhasib — CHANGES_REQUESTED** [review]
Lets close this PR

**Pull request closed without merging**